src/bin/typos-cli/report.rs

```diff
@@ -218,8 +220,11 @@ fn print_long_correction(msg: &Typo, palette: Palette) -> Result<(), std::io::Er
         let line_num = context.line_num.to_string();
         let line_indent: String = itertools::repeat_n(" ", line_num.len()).collect();
 
-        let hl_indent: String = itertools::repeat_n(" ", column).collect();
-        let hl: String = itertools::repeat_n("^", msg.typo.len()).collect();
+        let visible_column = UnicodeWidthStr::width(start.as_ref());
+        let visible_len = UnicodeWidthStr::width(msg.typo);
+
+        let hl_indent: String = itertools::repeat_n(" ", visible_column).collect();
+        let hl: String = itertools::repeat_n("^", visible_len).collect();
```

Comment:
> I've intentionally left column variable as is since many IDE counts by grapheme clusters.

I fully recognize I have no experience in this area and am just blindly trying to make things do the right thing  With that said, my couple of concerns
- This feels weird to me that we are rendering columns different than what we are telling people.  is there more where I can read on editors using graphemes and why?
- This feels weird for us to pull in a separate dependency for these two similar roles.

One thing that isn't a problem is the API.  It returns bytes which seems the most correct so people can choose what they want on top of this.

Reply:
I'm just a regular user of text editors 😄 sorry. I couldn't find a good one to read.
However, I can summerize some editor and compiler's behavior as a regular user. See [this comment](https://github.com/crate-ci/typos/pull/465#issuecomment-1109355495).
In short, Rust compiler and editors count column numbers in codepoints, while editors count a horizontal tab character according to indenting configuration. And the codepoints count, the UnicodeWidthStr width, the actual column width in display, keystorke count to navigate, edit, and the grapheme cluster count are incoherent.
I personally like grapheme cluster based width.